File: JavaTestApp/app/Global.java
import info.schleichardt.play2.basicauth.CredentialsFromConfChecker;
import play.GlobalSettings;
import play.api.mvc.Handler;
import play.mvc.Http;
import info.schleichardt.play2.basicauth.JAuthenticator;

public class Global extends GlobalSettings {
    private JAuthenticator authenticator = new JAuthenticator(new CredentialsFromConfChecker());

    @Override
    public Handler onRouteRequest(final Http.RequestHeader requestHeader) {
        final Handler defaultHandler = super.onRouteRequest(requestHeader);
        return authenticator.requireBasicAuthentication(requestHeader, defaultHandler);
    }
}

File: ScalaTestApp/app/Global.scala
import play.api.GlobalSettings
import play.api.mvc.RequestHeader
import info.schleichardt.play2.api.basicauth.CredentialsFromConfChecker
import info.schleichardt.play2.api.basicauth.Authenticator

object Global extends GlobalSettings {

  val requireBasicAuthentication = Authenticator(new CredentialsFromConfChecker)

  override def onRouteRequest(request: RequestHeader) = requireBasicAuthentication(request, () => super.onRouteRequest(request))
}
